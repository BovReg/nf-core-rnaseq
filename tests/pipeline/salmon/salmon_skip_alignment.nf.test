nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "salmon"
    tag "pipeline"

    test("SALMON (Skip Alignment & Pseudo-Alignment) (GSE110004)") {

        when {
            params {
                outdir         = "$outputDir"
                pseudo_aligner = "salmon"
                skip_alignment = true
                skip_pseudo_alignment = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/pipeline_info/samplesheet.valid.csv")).match("samplesheet") },
                { assert snapshot(path("$outputDir/bbsplit/RAP1_IAA_30M_REP1.stats.txt"),
                                path("$outputDir/bbsplit/RAP1_UNINDUCED_REP1.stats.txt"),
                                path("$outputDir/bbsplit/RAP1_UNINDUCED_REP2.stats.txt"),
                                path("$outputDir/bbsplit/WT_REP1.stats.txt")).match("bbsplit") },
                { assert new File("$outputDir/bbsplit/WT_REP2.stats.txt").exists() },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_cutadapt.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt")).match("multiqc") },
                { assert new File("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt").exists() },
                { assert !new File("$outputDir/salmon/").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_IAA_30M_REP1_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/RAP1_UNINDUCED_REP2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP1_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP1_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP2_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/WT_REP2_2.fastq.gz_trimming_report.txt").exists() }
            )
        }

    }

}
