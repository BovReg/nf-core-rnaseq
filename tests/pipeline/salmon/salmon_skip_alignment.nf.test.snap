{
    "samplesheet": {
        "content": [
            "samplesheet.valid.csv:md5,21387c48b43947403fb84f5e33be9e9b"
        ],
        "timestamp": "2023-05-31T21:21:39+0000"
    },
    "software_versions": {
        "content": [
            "{BBMAP_BBSPLIT={bbmap=39.01}, CAT_ADDITIONAL_FASTA={python=3.9.5}, CAT_FASTQ={cat=8.30}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, CUSTOM_GETCHROMSIZES={getchromsizes=1.16.1}, FASTQC={fastqc=0.11.9}, FQ_SUBSAMPLE={fq=0.9.1 (2022-02-22)}, GTF2BED={perl=5.26.2}, GUNZIP_ADDITIONAL_FASTA={gunzip=1.10}, GUNZIP_GTF={gunzip=1.10}, SALMON_QUANT={salmon=1.10.1}, SAMPLESHEET_CHECK={python=3.9.5}, TRIMGALORE={cutadapt=3.4, trimgalore=0.6.7}, UNTAR_SALMON_INDEX={untar=1.30}, Workflow={nf-core/rnaseq=3.12.0dev}}"
        ],
        "timestamp": "2023-05-31T21:21:39+0000"
    },
    "bbsplit": {
        "content": [
            "RAP1_IAA_30M_REP1.stats.txt:md5,a87292c6f7166096166a9015c2e63dac",
            "RAP1_UNINDUCED_REP1.stats.txt:md5,781e49024150a8313779c574b58ad035",
            "RAP1_UNINDUCED_REP2.stats.txt:md5,e455d0163bf2d988224b0c0092605213",
            "WT_REP1.stats.txt:md5,6eab6936804c1157d5e1c61c0562d7de"
        ],
        "timestamp": "2023-05-31T21:21:39+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_cutadapt.txt:md5,aac9581a5670cb55edf564f3d6c1f9a7",
            "multiqc_fastqc.txt:md5,68bdc9b37911a79484e0333e82679fa3"
        ],
        "timestamp": "2023-05-31T21:21:39+0000"
    }
}